include/libtorrent/alert_types.hpp

```diff
@@ -66,6 +66,15 @@ namespace libtorrent
 	// user defined alerts should use IDs greater than this
 	static const int user_alert_id = 10000;
 
+	static const int base_alert_id_counter = __COUNTER__ + 1;
```

Comment:
so if __COUNTER__ is not supported then constexpr compile counter can be used instead?